src/main/java/org/jitsi/jicofo/Main.java

```diff
@@ -62,17 +64,12 @@ public static void main(String[] args)
         ConfigUtils.PASSWORD_CMD_LINE_ARGS = "secret,user_password";
 
 
-        final Object exitSyncRoot = new Object();
+        ShutdownServiceImpl shutdownService = new ShutdownServiceImpl();
         // Register shutdown hook to perform cleanup before exit
-        Runtime.getRuntime().addShutdownHook(new Thread(() ->
-        {
-            synchronized (exitSyncRoot)
-            {
-                exitSyncRoot.notifyAll();
-            }
-        }));
+        Runtime.getRuntime().addShutdownHook(new Thread(shutdownService::beginShutdown));
         logger.info("Starting OSGi services.");
-        BundleActivator activator = startOsgi(exitSyncRoot);
+        BundleActivator activator = new DummyActivator();
+        OSGiLauncher launcher = startOsgi(activator);
```

Comment:
We have to still start OSGi for some things, right?  But the dummy activator does nothing, so what behavior is it we're relying on there?

Reply:
At first flance `OSGiLauncher.start(BundleActivator)` requires a non-null activator, so I added that without digging further. But I see there a `start()` with no arguments that I can use.

Reply:
Ugh, `start()` is private...

Reply:
Ok...but what is it that happens as part of `OSGiLauncher.start` that we still need? (Just curious)

Reply:
Starting all of our bundles, we still have a bunch:
https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/osgi/JicofoBundleConfig.java#L69

Reply:
Oh, and those still work even though the activator is a no-op?

Reply:
Yep 🤷

Reply:
I think I could move `WaitableBundleActivator` from the bundle config here, but I'd rather not fiddle with it